### PR TITLE
Major update

### PR DESCRIPTION
**Pull request by deanh**
Major work by @janmonschke to update and improve the SoundCloud JS SDK:

* Move from CoffeeScript to ES2015
* Use Promises instead of callbacks
* Secure HLS playback
* HTML5 uploads
* WebAudio recording
* Webpack builds
* Karma testing
* New examples

### CODE REVIEW

**Comment by deanh** [comment]
@sferik: Please review!

File: LICENSE.MD [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Recorderjs License (MIT)
+
+Copyright © 2013 Matt Diamond
```
Comment (janmonschke) [inline LICENSE.MD]: This should probably not be Matt Diamond ;)

File: LICENSE.MD [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Recorderjs License (MIT)
```
Comment (janmonschke) [inline LICENSE.MD]: We should add our license as well :)

**Comment by sferik** [comment]
\o/

**Pull request merged**